runtime/src/main/java/com/datastax/oss/quarkus/runtime/internal/session/QuarkusCqlSessionBuilder.java

```diff
@@ -29,17 +29,23 @@
 public class QuarkusCqlSessionBuilder
     extends SessionBuilder<QuarkusCqlSessionBuilder, QuarkusCqlSession> {
 
-  private final MetricRegistry metricRegistry;
-  private final EventLoopGroup mainEventLoop;
-  private final boolean useQuarkusNettyEventLoop;
-
-  public QuarkusCqlSessionBuilder(
-      MetricRegistry metricRegistry,
-      EventLoopGroup mainEventLoop,
-      boolean useQuarkusNettyEventLoop) {
+  private MetricRegistry metricRegistry;
+  private EventLoopGroup mainEventLoop;
+  private boolean useQuarkusNettyEventLoop;
+
+  public QuarkusCqlSessionBuilder withMetricRegistry(MetricRegistry metricRegistry) {
```

Comment:
Do you mind adding nullability annotations? It would be nice to be able to tell if all these items are required or not.